### ISSUE TITLE
False positive of Style/RedundantLineContinuation

### ISSUE THREAD

**Issue by mkllnk** [rubocop/rubocop#12793]
## Expected behavior

The following code is correct:
```rb
        ids = managed_permitted_ids | hubs_permitted_ids | hubs_permitting_ids \
          | producers_permitted_ids | producers_permitting_ids | managed_active_ids \
          | hubs_active_ids | producers_active_ids
```
```rb
          filter_scope("spree_orders.completed_at >= ?", params.completed_at_gt) \
            if params.completed_at_gt.present?
          filter_scope("spree_orders.completed_at <= ?", params.completed_at_lt) \
            if params.completed_at_lt.present?
```
## Actual behavior

This is corrected to invalid code:
```rb
        ids = managed_permitted_ids | hubs_permitted_ids | hubs_permitting_ids \
          | producers_permitted_ids | producers_permitting_ids | managed_active_ids 
          | hubs_active_ids | producers_active_ids
```
```rb
          filter_scope("spree_orders.completed_at >= ?", params.completed_at_gt) 
            if params.completed_at_gt.present?
          filter_scope("spree_orders.completed_at <= ?", params.completed_at_lt) 
            if params.completed_at_lt.present?
```

## Steps to reproduce the problem

Run Rubocop on the examples above.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.62.1 (using Parser 3.3.0.5, rubocop-ast 1.31.2, running on ruby 3.1.4) [x86_64-linux]
  - rubocop-rails 2.24.0
  - rubocop-rspec 2.27.1
```

The issue was first observed in version 1.61.0. A possible fix was merged but the new version is still offending:

* https://github.com/rubocop/rubocop/issues/12726
